File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/wallet/WalletEntity.java
package com.github.dedis.popstellar.repository.database.wallet;

import androidx.annotation.NonNull;
import androidx.room.*;

import com.github.dedis.popstellar.model.Immutable;

import java.util.ArrayList;
import java.util.List;

@Entity(tableName = "wallet")
@Immutable
public class WalletEntity {

  // The id is used to keep always at most one entry in the database
  @PrimaryKey
  @ColumnInfo(name = "id")
  private final int id;

  @ColumnInfo(name = "wallet_seed")
  @NonNull
  private final List<String> walletSeed;

  public WalletEntity(int id, @NonNull List<String> walletSeed) {
    this.id = id;
    this.walletSeed = walletSeed;
  }

  public int getId() {
    return id;
  }

  @NonNull
  public List<String> getWalletSeed() {
    return new ArrayList<>(walletSeed);
  }

  public String[] getWalletSeedArray() {
    return walletSeed.toArray(new String[0]);
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/wallet/WalletEntity.kt
package com.github.dedis.popstellar.repository.database.wallet

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.github.dedis.popstellar.model.Immutable

@Entity(tableName = "wallet")
@Immutable
class WalletEntity(
    // The id is used to keep always at most one entry in the database
    @field:ColumnInfo(name = "id") @field:PrimaryKey val id: Int,
    @field:ColumnInfo(name = "wallet_seed") private val walletSeed: List<String>
) {

  fun getWalletSeed(): List<String> {
    return ArrayList(walletSeed)
  }

  val walletSeedArray: Array<String>
    get() = walletSeed.toTypedArray()
}
